src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -1202,6 +1192,42 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
         return responseRows;
     }
 
+    private void addCommentsToResponseRow(FeedbackResponseAttributes response, FeedbackQuestionAttributes question,
+            InstructorFeedbackResultsResponseRow responseRow, List<String> responseGiverRecipientList) {
+        responseRow.setCommentsOnResponsesAllowed(true);
+        String giverNameAndTeam = bundle.getNameForEmail(response.giver);
+        String recipientNameAndTeam = bundle.getNameForEmail(response.recipient);
+
+        String giverTeam = bundle.getTeamNameForEmail(response.giver);
+        String recipientTeam = bundle.getTeamNameForEmail(response.recipient);
+
+        giverNameAndTeam = bundle.appendTeamNameToName(giverNameAndTeam, giverTeam);
+        recipientNameAndTeam = bundle.appendTeamNameToName(recipientNameAndTeam, recipientTeam);
+
+        List<FeedbackResponseCommentRow> comments =
+                buildResponseComments(giverNameAndTeam, recipientNameAndTeam, question, response);
+        if (!comments.isEmpty()) {
+            responseRow.setCommentsOnResponses(comments);
+        }
+        Map<FeedbackParticipantType, Boolean> responseVisibilityMap = getResponseVisibilityMap(question);
+        FeedbackResponseCommentRow addCommentForm = buildFeedbackResponseCommentAddForm(question, response,
+                responseVisibilityMap, giverNameAndTeam, recipientNameAndTeam);
+        responseRow.setAddCommentButton(addCommentForm);
+
+        int giverIndex = responseGiverRecipientList.indexOf(response.giver);
+        // For some type of questions like "TEXT" when response giver is a team, we save "student's Team" as response giver
```

Comment:
I think it is because of legacy data. In current feedback submission code, if it is team, team name is saved.